tests/test_finite_difference.py

```diff
@@ -47,8 +44,7 @@ def test_multi_indep(x_data, w_data):
     Tests the case with multiple components, multiple parameters and
     multiple independent variables
     '''
-    w, x, y, z = sf.variables('w, x, y, z')
-    a, b, c = sf.parameters('a, b, c')
+    w = sf.Variable('w')
     model = sf.Model({y: 3 * a * x**2 + b * x * w - c,
                       z: sf.exp(a*x - b) + c*w})
```

Comment:
Is there a fundamental reason why this case doesn't fit in `test_model`? (which is a bad name by the way).
Also, the only thing I see fundamentally different from the cases above is that there are multiple independent variables. As-is it should be possible to create a case with 1 dependent and 2 independent variables above. 
The fact that we need at least 4 variables to create a case with multiple dependent and multiple independent variables is a strong argument to add an additional variable to the automatic list. I'll leave it to you whether that's going to a `w` or `x2`, or....

Reply:
I'll add a fourth variable (`w`).
We could combine both methods if we unpack a dictionary with data-arrays into keyword arguments (like we did with the parameters). That shouldn't be a problem 👍

Reply:
I found a "problem" :D 
See...Because only this specific test uses two data-arrays, we have "tree-like-data" and could use [nested parametrized tests](https://stackoverflow.com/a/40826505) (not pretty in my opinion).

Because the tree-branch isn't deep we could call this specific model twice, like
```python
@pytest.mark.parametrize('init_data', [{'x': np.arange(10)/10}, {'x': 3}])
@pytest.mark.parametrize('init_model,init_par',
    [
        ...
        ({y: 3 * a * x**2 + b * x * w - c, z: sf.exp(a*x - b) + c*w},
            {'a': 3.5, 'b': 2, 'c': 5, 'w': np.arange(10)/10}),
        ({y: 3 * a * x**2 + b * x * w - c, z: sf.exp(a*x - b) + c*w},
            {'a': 3.5, 'b': 2, 'c': 5, 'w': 5}),
    ]
)
```
and merge both dicts with `**dict(init_data,**init_par)`.

Reply:
Note that `w` is a *variable*, not a parameter. I don't think it matters in this test, but please check.
I'm not a great fan of the nested parametrization either

Reply:
> Note that `w` is a *variable*

I added `w` as a variable to `auto_variables`. The part `'w': 5` exists in the old version of the test too.

Of the two possibilities handling the data-array of `w`, I would propose and implemented the second one (with the written code, not the StackOverflow answer).